packages/openapi-v3/src/__tests__/unit/consolidate-schema.unit.ts

```diff
@@ -0,0 +1,195 @@
+    };
+    expect(ConsolidateSchemaObjects(inputSpec)).to.eql(expectedSpec);
+  });
+
+  it('ignores schema without title property', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    };
+    expect(ConsolidateSchemaObjects(inputSpec)).to.eql(inputSpec);
+  });
+
+  it('Avoids naming collision', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          title: 'loopback.example',
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+      components: {
+        schemas: {
+          LoopbackExample: {
+            title: 'Different LoopbackExample exists',
+            properties: {
+              testDiff: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expectedSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          $ref: '#/components/schemas/LoopbackExample2',
+        },
+      },
+      components: {
+        schemas: {
+          LoopbackExample: {
+            title: 'Different LoopbackExample exists',
+            properties: {
+              testDiff: {
+                type: 'string',
+              },
+            },
+          },
+          LoopbackExample2: {
+            title: 'loopback.example',
+            properties: {
+              test: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    };
+    expect(ConsolidateSchemaObjects(inputSpec)).to.eql(expectedSpec);
+  });
+
+  it('If array items has no title, copy parent title if exists', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          myarray: {
+            title: 'my.array',
+            type: 'array',
+            items: {
+              properties: {
+                test: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+    const expectedSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          myarray: {
+            title: 'my.array',
+            type: 'array',
+            items: {
+              $ref: '#/components/schemas/MyArray',
+            },
+          },
+        },
+      },
+      components: {
+        schemas: {
+          MyArray: {
+            title: 'my.array',
```

Comment:
Let's discuss.

Should we extract the entire array schema in this case? I am guessing that your intention is to support an array of model instances (e.g. the return value of `find` method) by keeping the array definition inside the endpoint spec and moving the item schema to shared components. I see how that can be a better solution. 

When moving the array item schema only, I feel it's confusing to see `MyArray` (or `my.array`) name when the schema describes array items (not the array itself). I am proposing to append `.items` suffix to avoid such confusion.

```suggestion
            title: 'my.array.items',
```

As mentioned in https://github.com/strongloop/loopback-next/pull/4355#discussion_r363785139, when building custom titles, there are few constraints to keep in mind:

- If we don't use any separator, then we can create a naming conflict when the application provides a model with the same name as we have created.
- IIRC, `.` and `-` are not allowed in LoopBack model names, therefore it should be safe to use them as a separator.
- Some OpenAPI consumers (e.g. Microsoft Azure) restrict the subset of characters allowed in schema titles. In particular, Azure requires titles to match the regular expression `^[a-zA-Z0-9.-_]+$` (see https://github.com/strongloop/loopback-next/pull/3504 and https://github.com/strongloop/loopback-next/issues/1466).

Reply:
Good idea, will append `.items` as next step.

Reply:
> Should we extract the entire array schema in this case? I am guessing that your intention is to support an array of model instances

It seems like the norm to use reference for items from what I can see with schema transformers.  A particularly great one is https://www.apimatic.io/transformer/, where you can import and export from/to Openapi v3 json and there are some nice things like specifically adding depreciation=false on endpoints.  In an ideal world they'd contribute to loopback-next! :)